TheAmazingAudioEngine/AEAudioFilePlayer.m

```diff
@@ -324,16 +324,43 @@ static OSStatus renderCallback(__unsafe_unretained AEAudioFilePlayer *THIS,
     }
     
     uint32_t silentFrames = THIS->_startTime && THIS->_startTime > time->mHostTime
-        ? AESecondsFromHostTicks(THIS->_startTime - time->mHostTime) * THIS->_unitOutputDescription.mSampleRate : 0;
-    AEAudioBufferListCopyOnStack(scratchAudioBufferList, audio, silentFrames * THIS->_unitOutputDescription.mBytesPerFrame);
+    ? AESecondsFromHostTicks(THIS->_startTime - time->mHostTime) * THIS->_unitOutputDescription.mSampleRate : 0;
+    
+    // NOTE: AEAudioBufferListCopyOnStack was causing strange bugs such as audio data being written to the audio AudioBufferList WITHIN the silent region defined by silentFrames.
+    //    AEAudioBufferListCopyOnStack(scratchAudioBufferList, audio, silentFrames * THIS->_unitOutputDescription.mBytesPerFrame);
+    
+    // Create the scratchAudioBufferList that AEAudioBufferListCopyOnStack was attempting to create. Part 1: Create the pointer.
+    char bytesForAudioBufferList[sizeof(audio->mNumberBuffers) + audio->mNumberBuffers * sizeof(AudioBuffer)];
+    AudioBufferList *scratchAudioBufferList = (AudioBufferList*)bytesForAudioBufferList;
+    
+    // Create a new timestamp that represents the start time for the scratchAudioBufferList. Part 1: Create the pointer.
+    char bytesForAdjustedTimeStamp[sizeof(AudioTimeStamp)];
```

Comment:
I think you can just use `AudioTimeStamp adjustedTimeStamp;` since it's not a resizable struct?